src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs

```diff
@@ -138,9 +138,8 @@ internal static int OptimalMaxSpinWaitsPerSpinIteration
                 return 7;
             }
         }
-
 #if TARGET_UNIX || TARGET_BROWSER
-        internal WaitSubsystem.ThreadWaitInfo WaitInfo => _waitInfo;
+        internal WaitSubsystem.ThreadWaitInfo WaitInfo => Volatile.Read(ref _waitInfo) ?? AllocateWaitInfo();
```

Comment:
```suggestion
        internal WaitSubsystem.ThreadWaitInfo WaitInfo
        {
            return Volatile.Read(ref _waitInfo) ?? AllocateWaitInfo();
            
            WaitSubsystem.ThreadWaitInfo AllocateWaitInfo()
            {
                Interlocked.CompareExchange(ref _waitInfo, new WaitSubsystem.ThreadWaitInfo(this), null);
                return _waitInfo;
            }
        }
```
Nit: To keep the related code together and reduce number of ifdefs.

Reply:
(Also delete the other AllocateWaitInfo. github does not let me to do it as suggestion.)